backend/crates/eiffelvis_gen/src/event_set.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+
+use crate::event_type::{Event, Link, LinkTargets};
+
+#[derive(Default)]
+pub struct EventSetBuilder {
+    links: HashMap<String, Link>,
+    events: HashMap<String, Event>,
+}
+
+#[derive(Default, Debug)]
+pub struct EventSet {
+    pub links: HashMap<String, Link>,
+    pub events: HashMap<String, Event>,
+}
+
+impl EventSet {
+    pub fn build() -> EventSetBuilder {
+        EventSetBuilder::new()
+    }
+}
+
+impl EventSetBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn add_event_set(mut self, event_set: impl Into<EventSet>) -> Self {
+        let event_set: EventSet = event_set.into();
+        println!("{:#?}", event_set);
+        self.links.extend(event_set.links);
+        self.events.extend(event_set.events);
+        self
+    }
+
+    pub fn add_event(mut self, event: Event) -> Self {
+        self.events.insert(event.name.clone(), event);
+        self
+    }
+
+    pub fn add_link(mut self, link: Link) -> Self {
+        self.links.insert(link.name.clone(), link);
+        self
+    }
+
+    pub fn build(self) -> Option<EventSet> {
```

Comment:
Maybe we should check if this EventSet is possible to generate (e.g. impossible requirements)